### ISSUE TITLE
Array intersection with list?

### ISSUE THREAD

**Issue by kkmuffme** [vimeo/psalm#9732]
https://psalm.dev/r/72e828726e

While "bar" reports an error, "foo" does not.
"foo" should report the same error as "bar", if `&list` intersection is supported, otherwise give an error for invalid docblock?

**Comment by orklah** [comment]
Yeah, looking at the combined form, Psalm expects `array{0?: non-empty-string, args: array<array-key, mixed>, ...<array-key, mixed>}` for the first method and `array{args: array<array-key, mixed>, ...<int, non-empty-string>}` for the second

The first one results in a more permissive form that should not accept `hello` as an additional key
